Modals and DashAlert state

ackernaut opened the pull request:
Fixes #724

mattkrick commented on src/universal/components/Dashboard/DashModal.js:
```diff
@@ -107,19 +104,19 @@ const styleThunk = (theme, props) => ({
-  },
-
-  main: {
+  [ui.modalLayout[0]]: {
```
if we write the type out instead of referencing an array this would be easier to read

mattkrick commented on src/universal/modules/teamDashboard/components/Team/Team.js:
```diff
@@ -65,17 +66,29 @@ const settingsLinks = (teamId) => {
   const isSettings = router.isActive(`/team/${teamId}/settings`, false);
   initialValues.teamName = teamName;
   const DashHeaderInfoTitle = isSettings ? <EditTeamName initialValues={initialValues} teamName={teamName} teamId={teamId}/> : teamName;
+  const modalLayout = hasNotificationBar ? ui.modalLayout[1] : ui.modalLayout[0];
```
referencing the array isn't very meaningful (i know i set a bad example with the notifications varList, but that's mainly for payload size). could we instead say something like `hasNotificationBar ? ui.main : ui.mainHasNotificaitonBar

mattkrick commented on src/universal/containers/DashLayoutContainer/DashLayoutContainer.js:
```diff
@@ -81,15 +83,28 @@ export default class DashLayoutContainer extends Component {
+    if (activeMeetings.length > 0 || barType === TRIAL_EXPIRES_SOON || barType === TRIAL_EXPIRED) {
+      this.props.dispatch(notificationBarPresent(true));
+    } else {
+      this.props.dispatch(notificationBarPresent(false));
```
can we grab `hasNotificationBar` and ONLY call dispatch if nextProps.hasNotificationBar !== this.props.hasNotificationBar? Otherwise we'll be triggering a dispatch on every new prop that comes in

ackernaut replied:
Yes! Very helpful, thanks!

ackernaut replied:
@mattkrick Not sure how to set it the first time without next always equal to the current.

mattkrick replied:
@ackernaut here's the pattern i use:
- factor this out into its own method
- call that method from componentWillMount and componentWillReceiveProps
- from cWRP, pass in this.props and nextProps
- from cWM, pass in this.props as the nextProps param and {} for nextProps

Check out `MeetingContainer` for the pattern, i do this for `handleRedirects`.

mattkrick commented on src/universal/modules/notifications/ducks/notificationDuck.js:
```diff
@@ -0,0 +1,23 @@
+const NOTIFICATION_BAR_PRESENT = 'notifications/NOTIFICATION_BAR_PRESENT';
```
where does this bar appear? all dashboards? meeting? i think we may wanna refactor it into a dashboard duck. we already have an entity called notifications & having a notifications duck that doesnt pertain to it is gonna get confusing.

ackernaut commented:
Ok @mattkrick all changes have been made! This PR is ready.

mattkrick commented on src/universal/components/DashboardWrapper/DashboardWrapper.js:
```diff
@@ -4,9 +4,9 @@ import DashLayoutContainer from 'universal/containers/DashLayoutContainer/DashLa
+  const {children, dispatch, location, title} = props;
   return (
-    <DashLayoutContainer>
+    <DashLayoutContainer dispatch={dispatch}>
```
@ackernaut is this necessary?

mattkrick commented on src/universal/modules/userDashboard/components/UserDashboard/UserDashboard.js:
```diff
@@ -5,17 +5,18 @@ import {DragDropContext as dragDropContext} from 'react-dnd';
   const title = pathname === '/me' ? 'User Dashboard' : 'User Settings';
   return (
-    <DashboardWrapper location={pathname} title={title}>
+    <DashboardWrapper dispatch={dispatch} location={pathname} title={title}>
```
necessary?

mattkrick commented on src/universal/redux/makeReducer.js:
```diff
@@ -35,6 +36,7 @@ const appReducers = {
   [DEFAULT_AUTH_REDUCER_NAME]: auth,
   cashay: cashayReducer,
   form: formReducer.plugin(formPlugin),
+  notifications: notificationReducer,
```
let's refactor to 'dashboard' or something similar so we don't confuse the notification bar with notifications

ackernaut commented:
@mattkrick what do you think about `DashboardAlert` instead of `NotificationBar` since the content is not really a notification at all, just a cousin sometimes (when pertaining to billing, but not active meetings)...

ackernaut commented:
Ok @mattkrick third time’s the charm, eh?

- Refactored Noti…Bar to DashAlert (including ducks)
- Cleaned up unnecessary dispatch props
- Improved the DashAlert presence reducer to ensure bool value

mattkrick commented on src/universal/modules/newTeam/components/NewTeam/NewTeam.js:
```diff
@@ -10,7 +10,7 @@ import HTML5Backend from 'react-dnd-html5-backend';
   const {dispatch, params: {newOrg}} = props;
   return (
-    <DashboardWrapper title="User Dashboard">
+    <DashboardWrapper dispatch={dispatch} title="User Dashboard">
```
@ackernaut can we get rid of this?

ackernaut commented:
👍

jordanh commented:
Is this ready to merge now?
